src/components/Post.tsx

```diff
@@ -59,98 +108,146 @@ export default function Post({item}: PostProps) {
       easing: Easing.inOut(Easing.ease),
     });
   };
-
-  useEffect(() => {
-    animatedHeight.value = isShowMore
-      ? Math.min(contentHeight, maxHeroHeight)
-      : textHeight;
-    handleAnimation();
-  }, [isShowMore, contentHeight]);
-
-  const onTextLayout = (event: LayoutChangeEvent) => {
-    setContentHeight(event.nativeEvent.layout.height);
-  };
-
-  const onContainerPress = () => {
-    if (isShowMore) {
-      toggleShowMore();
-    }
-  };
-
   const navigation: any = useNavigation();
   const ImageInfo = createImageInfo(navigation, isLiked, setIsLiked);
 
-  const DescriptionView = () => {
-    if (!item.description) {
-      return null;
-    }
-    if (item.description.length > 120) {
-      return (
-        <ScrollView
-          showsVerticalScrollIndicator={false}
-          style={{flexGrow: 0}}
-          contentContainerStyle={{
-            flexGrow: 0,
-          }}>
-          <Pressable onPress={toggleShowMore}>
-            <Text
-              style={styles.description}
-              onLayout={isShowMore ? onTextLayout : undefined}>
-              {isShowMore
-                ? item.description
-                : `${item.description.slice(0, 120)}...`}
-              <Text style={styles.showText}>
-                {isShowMore ? ' Show Less' : ' Show More'}
-              </Text>
-            </Text>
-          </Pressable>
-        </ScrollView>
-      );
-    } else {
-      return (
-        <Text style={styles.description} onLayout={onTextLayout}>
-          {item.description}
-        </Text>
-      );
-    }
-  };
-
   return (
-    <Pressable onPress={onContainerPress} style={{height, flex: 1}}>
+    <Pressable onPress={onContainerPress} style={{height, width, flex: 1}}>
       <LinearGradient
         colors={['rgba(0, 0, 0, 0.8)', 'transparent']}
         style={styles.topGradient}
       />
+      <Video
+        onLoad={e => setDuration(e.duration)}
+        onProgress={e => {
+          setCurrentTime(e.currentTime);
+        }}
+        onPlaybackStateChanged={e => setIsPlaying(e.isPlaying)}
+        onEnd={handleVideoEnded}
+        source={{uri: item.videoSource}}
+        style={[StyleSheet.absoluteFill]}
+        resizeMode="cover"
+        ref={videoRef}
+        paused={paused}
+      />
+      {!isSeeking && paused ? (
+        <View
+          style={{
```

Comment:
use inline styles only for display please, move others to styles.whatevername as you use the same style here & replay button container